### PR TITLE
Fixed Datasource action SetValue issue for nextavail

### PR DESCRIPTION
**Pull request by arvindgh**

### CODE REVIEW

File: Ginger/GingerCoreNET/DataSource/LiteDB.cs [hunk 1]
```diff
@@ -708,6 +708,12 @@ public override bool RunQuery(string query)
         public void RunQuery(string query, int LocateRowValue, string DSTableName, bool MarkUpdate = false, bool NextAvai = false)
         {
             DataTable dt = GetQueryOutput("db." + DSTableName + ".find");
+            //Add a row if the table is empty to update the value.
+            if (dt.Rows.Count == 0 && NextAvai == true)
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/DataSource/LiteDB.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035&bid=16491792)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035&bid=16491792)

File: Ginger/GingerCoreNET/DataSource/LiteDB.cs [hunk 2]
```diff
@@ -719,9 +725,12 @@ public void RunQuery(string query, int LocateRowValue, string DSTableName, bool
             //Nextavailable
             else if (NextAvai)
             {
-                DataTable datatble = GetQueryOutput("db." + DSTableName + ".find GINGER_USED = \"False\" limit 1");
-                row = datatble.Rows[LocateRowValue];
-                string rowID = Convert.ToString(row["GINGER_ID"]);
+                DataTable datatble = GetQueryOutput("db." + DSTableName + ".select $.GINGER_ID where GINGER_USED = \"False\" limit 1");
+                row = datatble.Rows[0];
+                string rowID = row[0].ToString();
+                //DataTable datatble = GetQueryOutput("db." + DSTableName + ".find GINGER_USED = \"False\" limit 1");
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/DataSource/LiteDB.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035)

File: Ginger/GingerCoreNET/DataSource/LiteDB.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerCoreNET/DataSource/LiteDB.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035&bid=16491792)

File: Ginger/GingerCoreNET/DataSource/LiteDB.cs (hunk 2 above)
Comment (ambarkhan) [inline Ginger/GingerCoreNET/DataSource/LiteDB.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035)

File: Ginger/GingerCoreNET/DataSource/LiteDB.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerCoreNET/DataSource/LiteDB.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4997035)

**Comment by arvindgh** [comment]
No need to add row

**Pull request closed without merging**